Sign In With Apple Setup and Configuration

tamigoodall opened the pull request:
Sign In With Apple Setup and Configuration

https://docs-content-staging-pr-8191.herokuapp.com/docs/connections/social/apple

<!---
Notes:
- Your PR should conform to our [Contributing Guidelines](https://github.com/auth0/docs/blob/master/CONTRIBUTING.md)
- If applicable, add details to the [update feed](https://github.com/auth0/docs/tree/master/updates)
- Make sure your PR gets deployed to a Heroku [Review App](https://github.com/auth0/docs/blob/master/CONTRIBUTING.md#review-apps) without errors
- Please include a short description
- If your PR is a work in progress, title it [DO NOT MERGE]
--->

tamigoodall committed:
Sign In With Apple Setup and Configuration

Sign In With Apple Setup and Configuration

tamigoodall committed:
Update apple.md

fixed script text error

tamigoodall committed:
Update apple.md

Fixed broken indents

tamigoodall committed:
Update apple.md

Added table for variables

tamigoodall committed:
Minor edits and grammar

Minor edits and grammar

tamigoodall committed:
Merge branch 'master' into docs-875-native-siwa

tamigoodall committed:
Update apple.md

minor edits

rkhoriander committed:
Fix code sample breaking build

tamigoodall committed:
Moved text and added notes and screenshots

Moved text and added notes and screenshots

tamigoodall committed:
Merge branch 'docs-875-native-siwa' of github.com:auth0/docs into docs-875-native-siwa

tamigoodall committed:
Update apple.md

Updated flow description

tamigoodall committed:
Update apple.md

Fixed "Sign In with Apple" capitalization

tamigoodall committed:
Clarified instructions and links

Clarified instructions and links

tamigoodall committed:
Added new screenshot

Added new screenshot for Applications > Settings > Advanced > Mobile Apps
Removed "i.e." and "e.g."
Fixed indent

tamigoodall committed:
Typo fix

Typo fix

tamigoodall committed:
Update apple-app-mobile-settings.png

Replaced screenshot

josh-cain reviewed (commented):
Hi Tami!  Wanted to start the review process here so it's all in one place.  First, to answer the questions you posed in Slack:

> UI changes: Are they done yet? Can you give me a list of which Dashboard pages have changes? I will need new screenshots for Connections > Social Apple logo without “Beta”, > Settings, Applications > Advanced Settings > Mobile Settings

Yes, discussion and screenshots can be found here: https://auth0.slack.com/archives/CLJ3WSRDL/p1567633979029100?thread_ts=1567555222.021400&cid=CLJ3WSRDL

> Is there a list of known issues and limitations that I can include?

We do not have a canonical list, no.  Right now we've been working with customers (I.E. Headspace) to include several representative use cases (multi-apple platform, etc).  I'm sure we'll have some by launch time though.

> Are there any troubleshooting items and solutions yet?

Have not created them at this time.  Could you perhaps give an example of what you're looking for and we can provide troubleshooting helps?

> Are there any logs that we need to document?

Yes, we'll have tenant logs around items like:

- Apple code -> Auth0 token exchange occurrences
- Apple IdP interactions (failed exchange, error conditions, etc.)

We've captured the work for this here: https://auth0team.atlassian.net/browse/IPS-658, but it has not yet been picked up.

> The new doc (based on the Beta version) included a test procedure to validate the Apple connection using Auth0 Developer credentials. Do we still need to include that?

Not for native social.  Since the "AppID" is a unique identifier that is associated with a platform, we can't actually test native configurations from the Auth0 side.  Only web.

> If this affects tenant rate limitations, how? Is there anything written up about that yet? I saw a slack post about it on #native-social-login. (edited) 

We're in the process of finalizing the nuances of the rate limits for this here: https://auth0team.atlassian.net/wiki/spaces/IAMPS/pages/370280276/Native+Social+Login+Rate+Limits

josh-cain commented on articles/connections/social/apple.md:
```diff
@@ -18,49 +17,371 @@ useCase:
-* A [Custom Domain](/custom-domains) set up on your Auth0 tenant (this is required because you must be able to do domain verification with Apple).
+To set up and configure native Sign In with Apple, you will do the following:
+
+* [Complete the prerequisites](#before-you-begin): Have an Apple Developer account (paid), an Auth0 tenant set up with a custom domain, and a web application configured to use Auth0 for authentication at that domain.  
```
- Custom domains are not strictly required to use SIWA auth.  Domain validation is required for sending emails to private Apple addresses.
- Could be a web application *or* Native app.

josh-cain commented on articles/connections/social/apple.md:
```diff
@@ -18,49 +17,371 @@ useCase:
+To set up and configure native Sign In with Apple, you will do the following:
+
+* [Complete the prerequisites](#before-you-begin): Have an Apple Developer account (paid), an Auth0 tenant set up with a custom domain, and a web application configured to use Auth0 for authentication at that domain.  
+* [Register your app in your Apple Developer account](#register-your-app-in-your-apple-developer-account): Obtain Apple IDs and keys for the application connection settings in the Dashboard.
```
We should be explicit in what they'll need to configure here.  "Apple IDs" is ambiguous, they'll need:

- For web: an Apple ServiceID
- For Native: An Apple AppID (aka "Bundle Identifier")

josh-cain commented on articles/connections/social/apple.md:
```diff
@@ -18,49 +17,371 @@ useCase:
+* [Complete the prerequisites](#before-you-begin): Have an Apple Developer account (paid), an Auth0 tenant set up with a custom domain, and a web application configured to use Auth0 for authentication at that domain.  
+* [Register your app in your Apple Developer account](#register-your-app-in-your-apple-developer-account): Obtain Apple IDs and keys for the application connection settings in the Dashboard.
+* [Create your web app](#create-your-web-application): Including verifying your domain ownership with Apple and configuring Apple's email relay service, if necessary.
+* [Generate client secret](#generate-client-secret): Define the environment variables needed to use with Apple.
```
This feels like more of a run-time operation, rather than a setup/prerequisite.  The generated client secret has an expiration time (I think we use an hour), and is usually done at runtime.

josh-cain commented on articles/connections/social/apple.md:
```diff
@@ -18,49 +17,371 @@ useCase:
+
+## How it works
+
+Support for native Sign In with Apple is built on top of the [OAuth 2.0 Token Exchange specification](https://tools.ietf.org/html/draft-ietf-oauth-token-exchange-16). Auth0 created a profile for native SIWA to handle the following flow:
```
May want to call out that Auth0's *integration* with SIWA is built on that?  Just want to make clear the distinction that Apple uses standard(ish) OAuth 2 flows, and then we do token exchange AFTER (in addition to), in order to provide a way to hook into the Auth0 context.

josh-cain commented on articles/connections/social/apple.md:
```diff
@@ -18,49 +17,371 @@ useCase:
+
+Support for native Sign In with Apple is built on top of the [OAuth 2.0 Token Exchange specification](https://tools.ietf.org/html/draft-ietf-oauth-token-exchange-16). Auth0 created a profile for native SIWA to handle the following flow:
+
+1. User authenticates via the Apple SDK on their iPhone/iPad. They receive an authorization code.
```
May want to call out the distinction that the user no longer has to leave the app and use a browser.

josh-cain commented on articles/connections/social/apple.md:
```diff
@@ -18,49 +17,371 @@ useCase:
+    - `client_id`: their Auth0 Client ID
+    - `audience` and `scope` as needed (optional)
+3. Auth0 exchanges the `code` with Apple for a set of ID, access, and refresh tokens.
+4. Auth0 saves the user profile, executes rules and authorization, then issues access tokens (refresh tokens and ID tokens) as requested. These tokens are used to protect your APIs and users managed by Auth0.
```
Good sentence "These token are used..." !  Will help answer the inevitable "so I have the tokens, now what?" question.

josh-cain replied:
One thought: do we want to be *extra* clear about stating token source, I.E "auth0 access token" vs "apple access token"?  Could make it more difficult to read/parse, but the token provider may help dispel confusion as well.

josh-cain commented on articles/connections/social/apple.md:
```diff
@@ -18,49 +17,371 @@ useCase:
+Before you add support for native SIWA, you'll need:
+
+* An [Apple Developer](https://developer.apple.com/programs/) account, which is a paid account with Apple. (There is no free trial available unless you are part of their [iOS Developer University Program](https://developer.apple.com/support/compare-memberships/)).
+* A domain (such as `<YOUR CUSTOM DOMAIN>.com`) that you can use and point to your web app and an internet-accessible server where you will run the app, and that responds on behalf of this domain. You will also need to configure this server with a TLS certificate (Apple won't accept unsecured HTTP connections) and [`npm` and `Node.js`](https://nodejs.org/en/download/) (so you can run the web appl ...
```
- Same as above, custom domains aren't strictly required
- Might be native app, not web app
- "and [`npm` and `Node.js`]..." - Is this tailored to a quick-start, or is this intended to be general setup instructions?  If general, we need to drop the Node-specific pieces.

josh-cain commented on articles/connections/social/apple.md:
```diff
@@ -18,49 +17,371 @@ useCase:
+
+When setting up your application, make sure you save the following IDs and keys for the application connection settings in the Auth0 Dashboard:
+
+* Client ID (the Service ID)
```
May be AppID as well, depending.

josh-cain commented on articles/connections/social/apple.md:
```diff
@@ -18,49 +17,371 @@ useCase:
+
+6. Leave the other options with their default values and click **Continue**. When you click on this button, Apple  displays a summary of the options you just configured. If everything is correct, click **Register**. Apple redirects you to the **Certificates, Identifiers & Profiles** subsection again, this time listing your new App ID. 
+
+7. Create a Service ID that represents your application. This might look like a redundant effort, but Apple organizes it so you can nest multiple Services IDs under the same App ID. This makes sense when you have distinct versions of your application to support different devices.
```
These steps vary based on native vs web app

josh-cain reviewed (commented):
I think we might need to focus on a couple of concepts for our meeting, and how to best handle those in docs.  Namely:

- Difference between SIWA in a Web App vs Native App
- Use of domain validation / redirect URLS / email relays

tamigoodall commented:
@josh-cain Thanks for answering my questions and providing the links. 
Here are some more questions which you can answer here or during our call later today.
1. Do we need a separate set of instructions for web apps even though they aren't offered in the Apple App Store?
2. Do you think it makes sense to create an overview topic (including how it works and general stuff) for Native Social Authentication - then link it to the specific steps for Apple and potentially others in the future? This assumes that we will have more Native Social instructions down the line.
3. I'd like to add a use case or two if we can. Do you have one for Headspace and/or SkyScanner?
4. Is there an alternative set of instructions using the Mgmt API?

tamigoodall committed:
Broke content up

Broke content up

tamigoodall committed:
Fixed includes

Fixed includes

tamigoodall committed:
build fix

build fix

tamigoodall committed:
Fixed typo in filename

Fixed typo in filename

tamigoodall committed:
Added Keep reading Links

Added Keep reading list of links to related docs

tamigoodall committed:
Merge branch 'master' into docs-875-native-siwa

tamigoodall committed:
Delete unneeded file

Deleted verify-apple-connection-with-auth0-credentials
Moved content from add-siwa-to-web-app to generate-client-secret

tamigoodall committed:
Incorporated review comments

Incorporated review comments

tamigoodall committed:
Update sign-in-with-apple-overview.md

Removed large diagram and added placeholder

tamigoodall committed:
Build fix

Removed some metadata from subfolder files

tamigoodall committed:
Build fix - add apple.md back in

Added apple.md back into folder structure

tamigoodall committed:
Added overview back to apple.md

Added overview info
Removed ../ from include path

tamigoodall committed:
Fixed overview link

Fixed overview link

tamigoodall committed:
Moved files to references

Moved files to /connections/references

tamigoodall committed:
Updated links

Updated links

tamigoodall committed:
Update siwa-rate-limits.md

Added details

tamigoodall committed:
Update add-siwa-to-web-app.md

Interim edits - more to come

ziluvatar commented on articles/connections/social/apple.md:
```diff
@@ -18,49 +16,54 @@ useCase:
+2. The application calls Auth0's `/oauth/token` endpoint with the following parameters:
+    - `subject_token`: the authorization code they received above
+    - `subject_token_type`: `http://auth0.com/oauth/token-type/apple-authz-code`
+    - `grant_type`: `urn:ietf:params:oath:grant-type:token-exchange`
```
```suggestion
    - `grant_type`: `urn:ietf:params:oauth:grant-type:token-exchange`
```

ziluvatar commented on articles/connections/social/apple.md:
```diff
@@ -18,49 +16,54 @@ useCase:
+    - `grant_type`: `urn:ietf:params:oath:grant-type:token-exchange`
+    - `client_id`: their Auth0 Client ID
+    - `audience` and `scope` as needed (optional)
+3. Auth0 exchanges the `code` with the identity provider for a set of Auth0 ID, access, and refresh tokens.
```
a) I would not name it `code` since there wasn't any code in the parameters from step 2, maybe "`subject_token` (authorization code from Apple)".

b) Also for "identity provider" here is really Apple, maybe we can name it.

c) "for a set of Auth0 ID, access, and refresh tokens". Auth0 exchanges for a set of id token, access token and refresh token, which are from Apple. Let me know if "Auth0" ID was there for some reason I missed.

ziluvatar commented on articles/connections/social/apple.md:
```diff
@@ -18,49 +16,54 @@ useCase:
+    - `client_id`: their Auth0 Client ID
+    - `audience` and `scope` as needed (optional)
+3. Auth0 exchanges the `code` with the identity provider for a set of Auth0 ID, access, and refresh tokens.
+4. Auth0 saves the user profile, executes rules and authorization, then issues Apple access tokens (refresh tokens and ID tokens) as requested. These tokens are used to protect your APIs and users managed by Auth0.
```
> then issues Apple access tokens
In this case the tokens are from Auth0 not from Apple

ziluvatar commented on articles/connections/references/apple-native/guides/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,91 @@
+
+## Prerequisites
+
+<%= include('../_includes/_siwa-prerequisites') %>
```
It is not clear for me: if I only have a mobile app, do I have to use a custom domain? or can I still use the auth0.com domain?
I say it because it mentions the cases for the webapp, at the end I'm not sure if they apply to native too in some cases.

josh-cain replied:
You don't strictly need a custom domain in *any* case.  That's used for email relay validation.  Suppose your app doesn't send emails to end users, and just uses email as a primary for items like account linking - no domain configuration is required.

ziluvatar commented on articles/connections/references/apple-native/guides/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,91 @@
+
+13. Click **Download** for Apple to send you a file called `apple-developer-domain-association.txt`. You will have to use the contents of this file soon. Also, keep the page open because you will have to use the **Verify** button later.
+
+## Configure and test the connection in Auth0
```
For native apps it is needed to configure two things:
- Client > Show Advance Settings > Mobile Settings by enabling the toggle that the image below shows and the app id.
- Connection => which basically is this https://auth0.com/docs/connections/social/apple#2-create-and-enable-a-connection-in-auth0 and enable the connection for the client. 

Also not sure if the "Test Connection" makes sense because it would run a testing for a web app not for a native app. @josh-cain may have this clearer.

josh-cain replied:
That's correct, due to the way Apple issues AppID's to devices, *there is no way to test a native social connection* from the Auth0 server back-end.

ziluvatar commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,230 @@
+When setting up your application, make sure you save the following IDs and keys for the application connection settings in the Auth0 Dashboard:
+
+* Client ID (the Services ID) 
+* AppID
```
Don't we need client secret or signing key in this case @josh-cain?

josh-cain replied:
Yes.  I'd just use the same list from the other doc (in the same format):

- Client ID (the Service ID)
- Client Secret Signing Key
- Apple Team ID
- Client Signing Key ID (optional)

ziluvatar commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,230 @@
+    | **passport** | Use this package to handle user authentication in your app. |
+    | **passport-oauth** | Use this passport strategy to integrate with Apple's identity provider. |
+
+2. After installing these packages, create a new file called `app.js` and insert the following code into it:
```
In the JS example that web will sign in with apple, however, it won't use Auth0, is that right? Shouldn't they configure the connection and then do this while authenticating through Auth0?
@josh-cain

josh-cain replied:
I think there are bigger questions around this.  I think we should separate concerns of:

- How do I configure a SIWA web app?
- What would an example SIWA web app look like?

I think putting both in the same article can confuse users (I.E. which parts do I do if I've already got a web app?)

tamigoodall replied:
I've moved it to the end of the how-to guide and labeled it as an example.

ziluvatar commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
Are these specific values that Apple treat differently? just for domain verification or similar? @josh-cain

josh-cain replied:
Not sure I understand the question... these would be the SIWA-specific params used when calling the Apple IdP.

ziluvatar commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
This image should be one from connections. This I guess: https://auth0.com/docs/connections/social/apple#2-create-and-enable-a-connection-in-auth0

ziluvatar commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
Not sure if this can be reused: https://auth0.com/docs/dashboard/guides/connections/test-connections-social

ziluvatar commented on articles/connections/references/apple-native/references/siwa-rate-limits.md:
```diff
@@ -0,0 +1,33 @@
+  - customize-connections
+  - add-idp
+---
+# Sign In with Apple and Auth0 Rate Limits
```
I guess we may want to update https://auth0.com/docs/policies/rate-limits too

ziluvatar commented on articles/connections/references/apple-native/references/siwa-rate-limits.md:
```diff
@@ -0,0 +1,33 @@
+| Endpoint | Path | Limited by | Affected Tenants | Rate Limits |
+| --- | --- | --- | --- | --- |
+| Get Token	| `/oauth/token` | Native Social Login (Apple, Facebook, etc.) | Free |	100 requests per minute |
+|  |  | Native Social Login and IP | Paid |	500 requests per minute with burst of up to a 1000 requests |
```
These figures are not final yet. Pending on https://auth0team.atlassian.net/browse/IPS-632 reviews.

ziluvatar commented on articles/connections/references/apple-native/guides/generate-client-secret-siwa.md:
```diff
@@ -0,0 +1,112 @@
+---
+title: Generate Client Secret for Web Apps Using Sign In with Apple
```
I think I haven't seen a link to this page? 🤔

ziluvatar replied:
It's not clear to me what's the purpose of the page, the user using Auth0 should not care about the secret generation, right? 🤔

josh-cain replied:
Agree - I think we can give an example as a quickstart, but I'm very hesitant to document Apple's client secret generation on our site.

tamigoodall replied:
Ok - deleting this one

ziluvatar commented on articles/connections/references/apple-native/guides/test-siwa-connection.md:
```diff
@@ -0,0 +1,48 @@
+If you are using the Classic Universal Login flow, or embedding `Lock.js` in your application, make sure you are using `Lock.js` version 11.16 or later.
+:::
+
+1. Visit your application login page and you should see an option for Sign In with Apple:
```
In the case of native the starting point is not a website, it is a mobile app (tricky to test)

josh-cain commented on articles/connections/references/apple-native/_includes/_siwa-prerequisites.md:
```diff
@@ -0,0 +1,9 @@
+Before you add support for native SIWA, you'll need:
+
+* An [Apple Developer](https://developer.apple.com/programs/) account, which is a paid account with Apple. (There is no free trial available unless you are part of their [iOS Developer University Program](https://developer.apple.com/support/compare-memberships/)).
+* A domain (such as `<YOUR CUSTOM DOMAIN>.com`) that you can use and point to (for web apps) and an internet-accessible server where you will run the app, and that responds on behalf of this domain. You will also need to configure this server with a TLS certificate (Apple won't accept unsecured HTTP connections). Lastly, to use the Email Relay Service (for web apps only), you will need to configu ...
```
This prereq item has quite a bit of overlap with the next one - I'm not certain why we'd need both called out separately.

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,91 @@
+
+To set up and configure Sign In with Apple for your native app, you will do the following:
+
+1. [Complete the prerequisites](#before-you-begin): Have an Apple Developer account (paid), an Auth0 tenant set up with a custom domain, and a web application configured to use Auth0 for authentication at that domain.  
```
custom domains are not an Auth0 tenant requirement.  Non-custom domains would work, so long as your email relays are configured appropriately (if you even need those).

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,91 @@
+
+## Register your native app on your Apple Developer account
+
+Once your Apple Developer account is set up, follow the instructions in the [Apple Setup Guide](/connections/apple-setup) to set up your application. 
```
Note: Apple Setup Guide is web-app specific, yet referenced here in Native app setup guide.  Could be disorienting, as some of the steps do not match.

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,91 @@
+
+When setting up your application, make sure you save the following IDs and keys for the application connection settings in the Auth0 Dashboard:
+
+* Client ID (the Service ID)
```
(the App ID), thinking this will be listed in client settings as "App ID" to match terminology.

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,91 @@
+
+2. On the **Settings** tab, complete the fields **Client ID** (Services ID), **Client Secret Signing Key**, and **Apple Team ID**. You can also fill in the **Key ID** but it is optional, as Apple will accept the key without the ID.
+
+    ![Application Connection Settings: Advanced Mobile Settings](/media/articles/connections/social/apple/apple-app-mobile-settings.png)
```
This picture is a client settings dialog, but given in the context of connections settings instructions.

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,230 @@
+
+* [Complete the prerequisites](#prerequisites): Have an Apple Developer account (paid), an Auth0 tenant set up with a custom domain, and a web application configured to use Auth0 for authentication at that domain.  
+* [Register your app in your Apple Developer account](#register-your-app-in-your-apple-developer-account): Obtain Apple IDs and keys for the application connection settings in the Dashboard.
+* [Create your web app](#create-your-web-application): Including verifying your domain ownership with Apple and configuring Apple's email relay service, if necessary.
```
This step feels more like a quickstart.  Is this article aimed at a configuration task, or providing a quickstart sample to developers?

tamigoodall replied:
It is a how-to guide for configuring Sign In with Apple for a web app. The quickstarts are typically tutorials with sample code and we walk the developers through creating a working sample. We can remove the create app stuff here if that makes more sense.

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,230 @@
+
+2. When you sign into the developer account, go to the [Certificates, IDs, & Profiles](https://developer.apple.com/account/resources/certificates/list) section, and choose the [Identifiers](https://developer.apple.com/account/resources/identifiers/list/serviceId) subsection on the left menu.
+
+3. Click **Register an App ID** button.
```
We're in web app, so not App ID, Service ID

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,230 @@
+
+3. Click **Register an App ID** button.
+
+4. Choose **App IDs** as the identifier type, and click **Continue**. 
```
Service ID

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
Let's make sure the user knows this has to be done on the Apple developer portal, not the Auth0 portal.

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
May want to re-think this title.  Instructions do not cover how to generate the client secret, but rather information about client configuration.

josh-cain reviewed (commented):
Thanks for taking a shot at re-working this!  Some things I think went really well:

- Clear distinction between web and native flows
- Content in much more bite-sized chunks
- Content divisions feel right

However, I think some wires got crossed around:

 - difference between native and web configs (and associated terminology)
 - quickstart code vs configuration steps (both seem mixed)
 - when custom domains/verification are and are not needed

Happy to be as involved as I need to in order to get these items fixed and the docs wrapped up!

tamigoodall committed:
Update articles/connections/social/apple.md

Co-Authored-By: Eduardo Diaz <eduardo.diaz@auth0.com>

tamigoodall committed:
Incorported edits

Reworked prerequisites
Removed includes that don't make sense anymore
Fixed misplaced screenshots

tamigoodall committed:
Merge branch 'docs-875-native-siwa' of github.com:auth0/docs into docs-875-native-siwa

tamigoodall committed:
Incorporated reviewer comments (part 2)

Moved rate limits to main rate limits page (/policies/rate-limits.md)
Updated links
Deleted generate-client-secret-siwa.md
Deleted siwa-rate-limits.md
Rewrote main apple.md intro

tamigoodall committed:
Added siwa flow diagram

Added siwa flow diagram

tamigoodall committed:
Update apple.md

Fixed broken diagram link

tamigoodall committed:
Fixed config steps

Fixed config steps

tamigoodall committed:
Update apple.md

Fixed numbered list

tamigoodall committed:
Update add-siwa-to-native-app.md

Extra paren

jeffreylees commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,243 @@
+* A domain (such as <YOUR CUSTOM DOMAIN>.com) to point to and an internet-accessible server where you will run the app that responds on behalf of this domain. You will also need to configure this server with a TLS certificate (Apple won't accept unsecured HTTP connections). To use the Email Relay Service, you will also need to configure your domain with Sender Policy Framework (SPF) DNS TXT recor ...
+
+::: note
+You can set this up using [DigitalOcean](https://www.digitalocean.com/community/tutorials/how-to-install-nginx-on-ubuntu-18-04), [Freenom](https://freenom.com/), or [Let's Encrypt](https://letsencrypt.org/).
```
I'm not sure I would include this note. Setting up NGINX doesn't have much to do with the topic at hand. LetsEncrypt does, but as far as I know we don't recommend a certificate provider so it's probably best left at "You need a certificate".

jeffreylees commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,243 @@
+Before you add support for SIWA to your web app, you'll need:
+
+* An [Apple Developer](https://developer.apple.com/programs/) account, which is a paid account with Apple. (There is no free trial available unless you are part of their [iOS Developer University Program](https://developer.apple.com/support/compare-memberships/)).
+* A domain (such as <YOUR CUSTOM DOMAIN>.com) to point to and an internet-accessible server where you will run the app that responds on behalf of this domain. You will also need to configure this server with a TLS certificate (Apple won't accept unsecured HTTP connections). To use the Email Relay Service, you will also need to configure your domain with Sender Policy Framework (SPF) DNS TXT recor ...
```
As far as I know, this domain specifically needs to be registered as an Auth0 custom domain (the feature) and that seems to be lost in this refactor

josh-cain replied:
@jeffreylees - I'm not aware of this requirement.  I know there was some confusion early on about the use of customs domains (which were thought to be used for redirect URL validation, but instead are used for sender validation when using Apple's private MTA's).  Is there a reason why custom domains are required here?

jeffreylees replied:
No, if what you just said is the case, that answers that :)

jeffreylees commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,243 @@
+
+* A [Custom Domain](/custom-domains) set up on your Auth0 tenant for domain verification with Apple. Custom domains are not strictly required to use SIWA authorization. Domain validation is required for sending emails to private Apple addresses in native and web apps. 
+
+## Register your web app in your Apple Developer account
```
A general comment that when the registering an app with SIWA beta doc was written, the point was specifically to document Apple's end of the matter, including visuals, and screenshots are stripped out here. We don't normally document other products, especially not UIs, but it was intentional so just wanted to make sure that this sentiment has now changed.

jeffreylees replied:
Also, are the steps to register an app for native/web much different? Would there be any benefit to an include here, or no? I haven't done a step by step comparison yet.

josh-cain replied:
There are some significant differences between the two - namely the creation of a redirect URI whitelist for a web app, and the Service Identifier (web app) vs App ID distinction.

jeffreylees commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,243 @@
+
+When setting up your application with your Apple Developer account, make sure you save the following IDs and keys for the application connection settings in the Auth0 Dashboard:
+
+* Client ID (the Service ID)
```
Is the key no longer necessary for Web? If not, the above sentence no longer needs to say IDs and keys, I guess. Last time I checked it was still necessary though.

jeffreylees replied:
Actually, it's mentioned as needed further down the tutorial, so unsure why it's been removed

jeffreylees commented on articles/connections/references/apple-native/guides/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,91 @@
+
+9. Fill in the same fields as in step 4 above (description and identifier), and enable the **Sign In with Apple** feature. 
+
+10. Click **Configure**. Apple displays a dialog where you define the web domain you will use (`<YOUR CUSTOM DOMAIN>.com`) and add a **Return URL** (`https://<YOUR CUSTOM DOMAIN>/login/callback`). Click **Save**.  
```
Minor point, this should be <YOUR CUSTOM DOMAIN> not <YOUR CUSTOM DOMAIN>.com, as the extension is part of the custom domain, plus we don't know what the extension is. This is several places throughout the PR.

jeffreylees reviewed (commented):
This is what I could come up with for a quick skim. Hopefully that's helpful. Let me know if there's anything specific that you want double checked on Apple Dev.

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,91 @@
+
+6. Leave the other options with their default values and click **Continue**. When you click on this button, Apple  displays a summary of the options you just configured. If everything is correct, click **Register**. Apple redirects you to the **Certificates, Identifiers & Profiles** subsection again, this time listing your new App ID. 
+
+7. Create a Service ID that represents your application. This might look like a redundant effort, but Apple organizes it so you can nest multiple Services IDs under the same App ID. This makes sense when you have distinct versions of your application to support different devices.
```
This step is not strictly necessary to get an iOS app running.  Do we need to include it in a getting started style guide?

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,91 @@
+
+5. Click **Save**.
+
+4. [Test the connection](/dashboard/guides/connections/test-connections-social).
```
You can't test a native app from the web - won't work :(

josh-cain replied:
To be clear: if you followed the complete instructions set so far, it *would* work since you also configured a web component along with the native app.  However, I fear that app developers could be misled into thinking they were somehow testing their native app's connection here.

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,243 @@
+When setting up your application with your Apple Developer account, make sure you save the following IDs and keys for the application connection settings in the Auth0 Dashboard:
+
+* Client ID (the Service ID)
+* App ID
```
Don't need an App ID for a web app.  You will need the client secret signing key and (optionally) the key identifier)

josh-cain replied:
Well... let me walk that back a bit too.  You don't need *strictly* need an App ID for Auth0 configuration, and the only reason anyone ever has to have one is because Apple asks you to associate a Service Identifier with an App ID.  Still uncertain as to the the necessity of that link on the Apple side.

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,243 @@
+
+4. Choose **Service IDs** as the identifier type, and click **Continue**. 
+
+5. Enter a description for your new App ID and a Service ID. For the latter, Apple recommends using a reverse-domain name style string (for example, `com.<YOUR CUSTOM DOMAIN>.appid`). 
```
This doesn't make sense - in the Sevice ID creation screen, you're entering a description for a new Service ID, not App ID.

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,243 @@
+
+7. Leave the other options with their default values and click **Continue**. When you click on this button, Apple displays a summary of the options you just configured. If everything is correct, click **Register**. Apple redirects you to the **Certificates, Identifiers & Profiles** subsection again, this time listing your new App ID. 
+
+8. Click **Download** for Apple to send you a file called `apple-developer-domain-association.txt`. You will have to use the contents of this file soon. Also, keep the page open because you will have to use the **Verify** button later.
```
Download button is only available in the Redirect URI/domain validation dialog.  Not in the **Certificates, Identifiers, and Profiles** home.

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,243 @@
+
+5. Enter a description for your new App ID and a Service ID. For the latter, Apple recommends using a reverse-domain name style string (for example, `com.<YOUR CUSTOM DOMAIN>.appid`). 
+
+6. Scroll down and check the **Sign In with Apple** feature. You won't have to use the **Edit** here. You will configure this feature later.
```
We never revisit the steps needed to configure SIWA against a web app (redirect URI's).  Is that assumed from the web component of native configs in this doc?

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,243 @@
+
+1. On the Auth0 Dashboard, go to **Applications**. Click the gear to open the settings for your app. 
+
+2. At the bottom of the page, click **Show Advanced Settings** and go to the **Mobile Settings** tab. Under **Native Social Login**. enable the **Enable Sign In with Apple** toggle. 
```
We don't need to do this for a web app, can be enabled exclusively through the connections dialog.

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,243 @@
+
+6. [Test the connection](/connections/social/apple/guides/test-siwa-connection).
+
+## Web application configuration example
```
Still feels strange in a guide to me, but I'm no docs expert ;-)

josh-cain commented on articles/connections/references/apple-native/guides/add-siwa-to-web-app.md:
So far I haven't been asked for a callback when creating an Apple key...

josh-cain reviewed (commented):
- media/articles/connections/social/apple/apple-login-page.png - Why is the Apple logo not shown?

tamigoodall committed:
Incorporated review changes

Restructured files
Moved files
Renamed files
Split up files

tamigoodall committed:
Moved files and fixed links

Moved files and fixed links
Removed redundant text

tamigoodall committed:
Cleanup

Cleanup
Updated rate limits

tamigoodall committed:
Fixed dashboard links and misc typos

Fixed dashboard links
Fixed IDs and key names
Moved notes and screenshots
Removed unneeded Keep reading links

tamigoodall committed:
Last remaining comments

Fixes for last remaining comments

tamigoodall committed:
Added info on renewing authorization

Added info on renewing authorization

tamigoodall committed:
Link fix

Link fix

tamigoodall committed:
Delete siwa-concepts.png

Removed placeholder diagram

josh-cain commented on articles/connections/social/apple.md:
```diff
@@ -4,63 +4,65 @@ connection: Apple
 
-![Apple Connection Settings](/media/articles/connections/social/apple/apple_connection.png)
+* Call `ASAuthorizationAppleIDProvider.getCredentialState` to make sure the user is still authorized.
+* Retrieve Auth0 credentials.
```
This feels strange here.  Above ^, we don't go into detail around the code behind how a user performs the initial login.  Why do we do so here?  I think either both of these should go into that level of detail, or neither.  Feels like apples and oranges.

tamigoodall replied:
Agreed. I'll remove the details from the second case and just summarize it.

josh-cain commented on articles/connections/apple-siwa/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,75 @@
+
+For a native app, the Sign in with Apple login flow works as follows:
+
+![Sign In with Apple Authentication Flow](/media/articles/connections/social/apple/apple-siwa-authn-flow.png)
```
Some notes on this diagram:

- Step #3: only the auth code (no tokens) are sent from the app to Auth0

tamigoodall replied:
This is for Sonny - will resolve on our call.

josh-cain commented on articles/connections/apple-siwa/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,75 @@
+![Sign In with Apple Authentication Flow](/media/articles/connections/social/apple/apple-siwa-authn-flow.png)
+
+* User authenticates via Apple's SDK on their iPhone or iPad. They receive an authorization code. The user does not have to leave the app and use a browser to log in.
+* The application calls Auth0's `/oauth/token` endpoint with the following parameters:
```
+1 for including this - had a note to make sure it was called out for those who aren't using the Swift SDK.  Nice!

josh-cain commented on articles/connections/apple-siwa/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,75 @@
+    Auth0 exchanges the `subject_token` (authorization code) with Apple for an ID token, access token, and refresh token from Apple.
+* Auth0 saves the user profile, executes rules and authorization, then issues Auth0 access tokens (refresh tokens and ID tokens) as requested. These tokens are used to protect your APIs and users managed by Auth0.
+
+## Prerequisites
```
So - just curious.  We mention custom domain/email purposes in connection settings, but don't list that as a prerequisite here.  Email relay configs are the same (w/ domain validation) for web app configs as they are for native.  Not sure why it's one place and not the other.  Could we look to make the "Prerequisites" sections a little more consistent with one another?

tamigoodall replied:
Agreed - I'll make it more consistent.

josh-cain commented on articles/connections/apple-siwa/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,75 @@
+
+Once you have the credentials you need from your Apple Developer account, you need to configure the application client and the connection settings in Auth0.
+
+1. Go to [Dashboard > Applications > Settings](${manage_url}/#/applications/${account.clientId}/settings).  
```
Dashboard > Applications > [YOUR_APPLICATION] > Settings

(or however you want to represent clicking on the application)

josh-cain commented on articles/connections/apple-siwa/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,75 @@
+
+1. Go to [Dashboard > Applications > Settings](${manage_url}/#/applications/${account.clientId}/settings).  
+
+2. At the bottom of the page, click **Show Advanced Settings** and go to the **Mobile Settings** tab. Under **Native Social Login**. enable the **Enable Sign In with Apple** toggle. 
```
Have one more required step for this flow:

3. Under **iOS**, fill in the **App ID** field with the native app's App ID/Bundle Identifier

If they don't complete that ^ step, it won't work.

CriGoT replied:
Also `Mobile Settings` is being renamed `Device Settings`

josh-cain commented on articles/connections/apple-siwa/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,75 @@
+
+4. On the **Settings** tab, fill in the following fields:
+
+    * **Client ID** (Services ID)
```
Client ID not strictly necessary for native social - maybe mark as optional?

josh-cain commented on articles/connections/apple-siwa/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,75 @@
+
+    * **Client ID** (Services ID)
+    * **Client Secret Signing Key**
+    * (optional) **Key ID** (Apple will accept the key without the ID)
```
Note: they're going to need a **Team ID** in at least one of these places (Mobile Settings or Connection Settings).  I'd recommend instructing them to fill it out in connection settings.

josh-cain commented on articles/connections/apple-siwa/test-siwa-connection.md:
```diff
@@ -0,0 +1,43 @@
+
+::: note
+If you are using the Classic Universal Login flow, or embedding `Lock.js` in your application, make sure you are using `Lock.js` version 11.16 or later.
+:::
```
Might also want to add a note here about how native apps cannot be tested with browser-based flows?

tamigoodall replied:
I actually added it to the Native article instead.

josh-cain commented on articles/connections/apple-siwa/configure-email-relay-service.md:
```diff
@@ -0,0 +1,32 @@
+  - customize-connections
+  - configure-email-relay-service
+---
+## Configure Email Relay Service for Sign In with Apple
```
This page is simply hosting instructions for Apple.  Are we in the business of doing that?  If so, some pictures might help.  LMK if you want me to snag some.

tamigoodall replied:
I think we should delete this doc and just note that it needs to be done.

josh-cain commented on articles/connections/apple-siwa/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,77 @@
+
+  * Service ID (Client ID)
+  * Client Secret Signing Key
+  * App ID (Apple Team ID)
```
App ID != Apple Team ID

josh-cain commented on articles/connections/apple-siwa/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,77 @@
+
+## Verify domain ownership on the Apple Developer portal
+
+1. Move your project code to your server and make it run like you did locally. You still don't need to provide the final environment variables. To achieve that, you can use whatever means you prefer. For example, you can use Git, or you can move the files manually (with the help of `scp` or similar). 
```
"your project" and "provide the final environment variables" is really quickstart-y.  These statements don't make sense outside of that context.

josh-cain commented on articles/connections/apple-siwa/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,77 @@
+
+4. [Test the connection](/connections/social/apple/test-siwa-connection). 
+
+## Verify domain ownership on the Apple Developer portal
```
Remember: this step is still (to my knowledge...) not strictly necessary for setting up an app.

CriGoT replied:
+1 this is not required and it will not be possible for auth0 domains (tenant.auth0.com) and custom domains using auth0 managed certs.

josh-cain commented on articles/connections/apple-siwa/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,77 @@
+
+2. After running this project on an internet-accessible server (which must respond on behalf of the domain you configured in the Apple developer account), go back to the page you left open on the Apple Developer portal (**Register a Services ID**), and click **Verify**. If you got everything right, Apple will confirm that you own the informed domain.
+
+## Configure client variables
```
Same with this section.  "Configure client variables" is a section that's specific to a quickstart/implementation and does not belong in general connection config.

josh-cain reviewed (dismissed):
I'm liking the new structure, the HTTP call params, and the rate limiting docs.

Have some further clarifications/feedback that needs to be addressed though.  Getting close!

CriGoT commented on articles/policies/rate-limits.md:
```diff
@@ -319,3 +319,17 @@ For database connections, Auth0 limits certain types of repeat login attempts de
 
 There's a limit of 10 SMS messages/hour per user for multi-factor authentication. For more information, see [Configuring Twilio for SMS](/multifactor-authentication/twilio-configuration).
+
+## Limits on Sign In with Apple 
```
can this be renamed to Limits on "Native Social Login" or "Native Sign In with Apple"?
The current tifle might lead to some confusion since it does not apply to sign in with apple when using the web flow

CriGoT commented on articles/policies/rate-limits.md:
```diff
@@ -319,3 +319,17 @@ For database connections, Auth0 limits certain types of repeat login attempts de
+
+| Endpoint | Path | Limited by | Affected Tenants | Rate Limits |
+| --- | --- | --- | --- | --- |
+| Get Token	| `/oauth/token` | Native Social Login (Apple, Facebook, etc.) | Free |	30 requests per minute |
```
The path is being split in multiple lines which is not consistent with other tables in the same page

CriGoT commented on articles/connections/apple-siwa/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,75 @@
+
+2. At the bottom of the page, click **Show Advanced Settings** and go to the **Mobile Settings** tab. Under **Native Social Login**. enable the **Enable Sign In with Apple** toggle. 
+
+    ![Application Client Settings: Advanced Mobile Settings](/media/articles/connections/social/apple/apple-app-mobile-settings.png)
```
It should be updated with https://github.com/auth0/manhattan/pull/944 to reflect the change in the tab

CriGoT commented on articles/connections/apple-siwa/configure-email-relay-service.md:
```diff
@@ -0,0 +1,32 @@
+## Configure Email Relay Service for Sign In with Apple
+
+1. To use the email relay service that Apple provides, click **More** under the **Certificates, Identifiers, & Profiles** section and click **Configure**.
+2. Add your domain (`<YOUR CUSTOM DOMAIN>.com`) in the field available on **Domains and Associated Email Addresses**.
```
Mentioning Custom Domain here can be confusing because they might assume if the auth0 custom domain. I would rather be explicit saying:
```suggestion
2. Add the domain of the email addresses you will use to contact users (`<YOUR EMAIL DOMAIN>.com`) in the field available on **Domains and Associated Email Addresses**.
```

tamigoodall replied:
I've removed this docs completely.

CriGoT commented on articles/connections/apple-siwa/configure-email-relay-service.md:
```diff
@@ -0,0 +1,32 @@
+1. To use the email relay service that Apple provides, click **More** under the **Certificates, Identifiers, & Profiles** section and click **Configure**.
+2. Add your domain (`<YOUR CUSTOM DOMAIN>.com`) in the field available on **Domains and Associated Email Addresses**.
+3. Click **Register**.
+4. Add an email address (for example, `me@<YOUR CUSTOM DOMAIN>.com`) in **Individual Email Addresses**.
```
Same comment 
```suggestion
4. Add an email address (for example, `me@<YOUR EMAIL DOMAIN>.com`) in **Individual Email Addresses**.
```

tamigoodall replied:
I've removed this doc.

CriGoT commented on articles/connections/apple-siwa/configure-email-relay-service.md:
```diff
@@ -0,0 +1,32 @@
+3. Click **Register**.
+4. Add an email address (for example, `me@<YOUR CUSTOM DOMAIN>.com`) in **Individual Email Addresses**.
+5. Click **Register**.
+
```
I would add a note with the following

If you are using Auth0 email templates ( ie. Welcome email) make sure that the `from` email address domain is registered...

tamigoodall replied:
I've deleted this doc.

CriGoT commented on articles/connections/social/apple.md:
```diff
@@ -4,63 +4,65 @@ connection: Apple
-::: note
-You can test out the Apple connection by simply going to the [Dashboard](${manage_url}) to **Connections > Social** and Try the Apple connection, leaving the settings blank. This will let you test it out with Auth0’s developer credentials. Prior to use in production applications, however, you will need to set up your own, which this guide details.
+::: warning
+The [Apple App Store Developer Guidelines](https://developer.apple.com/app-store/review/guidelines/#sign-in-with-apple) require that SIWA must be available in all applications that use third-party sign-in options such as Facebook or Google.
```
```suggestion
The [Apple App Store Developer Guidelines](https://developer.apple.com/app-store/review/guidelines/#sign-in-with-apple) require that SIWA must be available in all applications that exclusively use third-party sign-in options such as Facebook or Google.
```

CriGoT commented on articles/connections/apple-siwa/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,77 @@
+
+* Have an [Apple Developer](https://developer.apple.com/programs/) account, which is a paid account with Apple. (There is no free trial available unless you are part of their [iOS Developer University Program](https://developer.apple.com/support/compare-memberships/)).
+
+* A [custom domain](/custom-domains) set up on your Auth0 tenant (because you must be able to do domain verification with Apple).
```
Unless they are using the Auth0 custom domain as the domain in the email addresses used to contact customers this is not true.
Just to be clear, a Web Domain does not have to be verified for the web flow to work. Customers can put the auth0 (tenant.auth0.com) or custom domain (customdomain.com) in the web domain without verification and everything works.
Also, custom domains using Auth0 managed certs cannot be verified because we don't give any option to upload the files.

CriGoT commented on articles/connections/apple-siwa/set-up-apple.md:
```diff
@@ -0,0 +1,127 @@
+
+After you register your application, you will be given the following IDs and keys to use in the application connection settings in the Dashboard:
+
+* Apple Team ID (App ID)
```
As mentioned by @josh-cain 

> App ID != Apple Team ID

CriGoT commented on articles/connections/apple-siwa/set-up-apple.md:
```diff
@@ -0,0 +1,127 @@
+
+* A paid [Apple Developer](https://developer.apple.com/programs/) account. 
+
+* A [Custom Domain](/custom-domains) set up. Custom domains are required for domain verification with Apple.
```
It is not.  the custom domain is only used as a redirect URI and that does not have to be verified the flow works without any verification.

CriGoT commented on articles/connections/social/apple.md:
```diff
@@ -4,63 +4,65 @@ connection: Apple
-* The Client Secret Signing Key
-* The Apple Team ID
-* The Client Signing Key ID (optional)
+* A [Custom Domain](/custom-domains) set up on your Auth0 tenant for domain verification with Apple. Custom domains are not strictly required to use SIWA authorization. Domain validation is required for sending emails to private Apple addresses in native and web apps. To use the [Email Relay Service](/connections/apple-siwa/configure-email-relay-service), you will also need to configure your doma ...
```
This is confusing and the two topics are independant
1. Custom domains is not required mainly because Web Domain verification is not required for an application to work. This is not required.
2. Domain validation for sending emails should not have anything to do with Custom Domains this is related to the domains used in email communications.

CriGoT commented on articles/connections/apple-siwa/set-up-apple.md:
```diff
@@ -0,0 +1,127 @@
+
+4. Click **Save**, **Continue**, and then click **Register**.
+
+### Verify domain with Apple
```
This is not required for everything to work

tamigoodall committed:
Update apple-siwa-authn-flow.png

Updated apple-siwa-authn-flow diagram from Sonny

tamigoodall committed:
Incorporate review comments

Incorporate review comments

tamigoodall committed:
Delete ERS setup and changed rate limit header

Deleted Apple Email Relay Service setup
Changed Rate Limit header title and links

tamigoodall committed:
Update apple-app-mobile-settings.png

New screenshot

tamigoodall committed:
Update articles/connections/social/apple.md

Co-Authored-By: Cristofer Gonzales <cristofer_g@hotmail.com>

tamigoodall committed:
Minor fixes

Minor fixes

tamigoodall committed:
Merge branch 'docs-875-native-siwa' of github.com:auth0/docs into docs-875-native-siwa

tamigoodall committed:
Typo and new diagram

Typo and new diagram

ziluvatar commented on articles/connections/apple-siwa/add-siwa-to-web-app.md:
```diff
@@ -0,0 +1,60 @@
+
+## Configure and enable the connection in Auth0
+
+Once you have the credentials you need from your Apple Developer account, you need to configure the application client and the connection settings in Auth0.
```
```suggestion
Once you have the credentials you need from your Apple Developer account, you need to configure the connection settings and enable the connection for your application in Auth0.
```

ziluvatar commented on articles/connections/apple-siwa/add-siwa-to-native-app.md:
```diff
@@ -0,0 +1,87 @@
+---
+# Add Sign In with Apple to Native iOS Apps
+
+You can add functionality to your native iOS application that allows your users to use Sign In with Apple to authenticate. In addition, you can also try the Auth0 [iOS Swift - Sign In with Apple Quickstart](/quickstart/native/ios-swift-siwa).
```
Link does not work. Is it something that will be there by the time of merging?

tamigoodall replied:
Yes

ziluvatar commented on articles/policies/rate-limits.md:
```diff
@@ -319,3 +319,36 @@ For database connections, Auth0 limits certain types of repeat login attempts de
+    <td>30 requests per minute</td>
+  </tr>
+  <tr>
+    <td>Tenant + IP</td>
```
```suggestion
    <td>Native Social Login request and IP</td>
```
This information is for a customer, so for themselves there is no concept of tenant + ip, only IP, so that I, as a customer, am limited by IP

ziluvatar commented on articles/policies/rate-limits.md:
```diff
@@ -319,3 +319,36 @@ For database connections, Auth0 limits certain types of repeat login attempts de
+  <tr>
+    <td rowspan="3">Get Token</td>
+    <td rowspan="3">/oauth/token</td>
+    <td>Native Social Login (Apple, Facebook, etc.)</td>
```
```suggestion
    <td>Any Native Social Login request</td>
```
I wouldn't add details about Facebook yet.

ziluvatar commented on articles/connections/apple-siwa/set-up-apple.md:
```diff
@@ -0,0 +1,124 @@
+
+    `node generate-secret.js`
+
+    This gives you the last of the items you need to provide when setting up the connection in Auth0's dashboard.
```
I am confused, the `app_secret` that we use to generate the `client_secret` sent to apple on the exchange:
https://github.com/auth0/auth0-server/blob/4c8b6ba8e9d8bb43553586249e7b09a98e709072/lib/auth/strategies/apple.js#L37

Is that really the token resulted from running that script? How is the ES alg verification working with that on their end? 🤔 

Easy to read: https://docs-content-staging-pr-8191.herokuapp.com/docs/connections/apple-siwa/set-up-apple#set-up-signing-key

cc: @josh-cain @panva

josh-cain replied:
It's not, had it removed, but we got our PR's crossed.  Apologies for the confusion.

ziluvatar reviewed (dismissed):
We are almost there!! 🎉

josh-cain committed:
remove custom domain requirement, tweak lists of prereqs and params

panva commented on articles/connections/apple-siwa/set-up-apple.md:
```diff
@@ -0,0 +1,125 @@
+4. Copy or move the `apple-developer-domain-association.txt` file to your application server and make it accessible at `https://customdomain.com/.well-known/apple-developer-domain-association.txt`.
+1. When this is ready, click **Verify** on your Services ID configuration page. If no error message is shown, your domain has been successfully verified.
+
+## Set up Signing Key
```
This section (Set up Signing Key) should only make the developer download the private key from apple, copy its ASCII content to clibpoard and paste it in the manage dashboard. No JWT signing to get the secret value, thats all handled by the connection in auth0 using the private key.

tamigoodall replied:
I believe Josh will be removing this section.

josh-cain replied:
Yeah, should be gone now.  Thanks 👍

josh-cain committed:
remove additional prereqs beyond apple developer account from connection landing page

tamigoodall committed:
Update articles/connections/apple-siwa/add-siwa-to-web-app.md

Co-Authored-By: Eduardo Diaz <eduardo.diaz@auth0.com>

tamigoodall committed:
Apply suggestions from code review

Co-Authored-By: Eduardo Diaz <eduardo.diaz@auth0.com>

josh-cain committed:
remove unneccesary sections from set up apple

josh-cain committed:
add authz api docs for TX endpoint

josh-cain committed:
tweak some verbiage on auth API description for TX + native social

josh-cain committed:
remove API params from native, link diagram explanation steps with numbers from picture

josh-cain committed:
add info on correct logout from native apps

tamigoodall committed:
Fixed link text

Fixed link text

tamigoodall committed:
Merge branch 'docs-875-native-siwa' of github.com:auth0/docs into docs-875-native-siwa

ziluvatar commented:
@josh-cain @tamigoodall it seems we removed the "Setup your Signing Key", I guess some steps are still needed so that the admin can get the client secret signing key.
From prod https://auth0.com/docs/connections/apple-setup#setup-your-signing-key 1-5 are valid, then download the key and copy and paste into our connection field, would that be right Josh?
Link to pr content: https://docs-content-staging-pr-8191.herokuapp.com/docs/connections/apple-siwa/set-up-apple

BTW location for that content has changed, will the link in the connection settings screen work well?

rkhoriander committed:
Punctuation fix

rkhoriander committed:
Minor punctuation fixes.

rkhoriander committed:
Minor punctuation fix

tamigoodall committed:
Replace setup flow diagram and fix typos

Replaced setup flow diagram
Minor typos fixed

tamigoodall committed:
Merge branch 'docs-875-native-siwa' of github.com:auth0/docs into docs-875-native-siwa

rkhoriander committed:
Minor punctuation/grammar edits

rkhoriander committed:
Minor punctuation edits

tamigoodall committed:
Fixed mislabeled link

Fixed link text from "mobile settings" to "device settings"

tamigoodall committed:
Merge branch 'docs-875-native-siwa' of github.com:auth0/docs into docs-875-native-siwa

rkhoriander committed:
Punctuation/consistency updates

rkhoriander committed:
Fix minor typos

rkhoriander committed:
Consistency, punctuation, grammar edits

rkhoriander committed:
Consistency edit

rkhoriander committed:
Punctuation/clarity edits

rkhoriander committed:
Update add-siwa-to-web-app.md
